libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java

```diff
@@ -438,6 +361,65 @@ public void stopNavigation() {
     }
   }
 
+  /**
+   * Initializes the offline data used for fetching offline routes.
+   * <p>
+   * This method must be called before {@link MapboxNavigation#findOfflineRouteFor(Point, Point, Point[])} /
+   * {@link MapboxNavigation#findOfflineRouteFor(Location, Point, Point[])}.
+   *
+   * @param tilesDirPath        path to directory containing tile data
+   * @param translationsDirPath path to directory containing OSRMTI translations
+   */
+  public void initializeOfflineData(String tilesDirPath, String translationsDirPath) {
+    mapboxNavigator.configureRouter(tilesDirPath, translationsDirPath);
+  }
+
+  /**
+   * Once {@link MapboxNavigation#initializeOfflineData(String, String)} has been called, you may
+   * call this method to retrieve routes without internet connectivity.
+   *
+   * @param origin      origin {@link Point} of the route
+   * @param destination destination {@link Point} of the route
+   * @param waypoints   way {@link Point}s of the route - these are optional
+   * @return an offline <tt>DirectionsRoute</tt>, <tt>null</tt> if an error is occurred fetching offline route or if
+   * storage permissions are not granted
+   */
+  @Nullable
+  public DirectionsRoute findOfflineRouteFor(@NonNull Point origin, @NonNull Point destination,
+                                             @Nullable Point... waypoints) {
```

Comment:
The idea behind using `varargs` here was the "optionality" that gives. Note the usage here https://github.com/mapbox/mapbox-navigation-android/blob/53b1e2a2ca2bd659002c40643214dace4084efd9/app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/OfflineRerouteActivity.java#L95 we're not passing in waypoints, only `origin` and `destination` are mandatory. I was trying to avoid clients of the API to pass in `null` when they're not adding waypoints, making clear that those are optional and do not include `origin` and `destination`. If a client wants to add waypoints they can add them as an array or as a sequence of arguments. If the client has a `List<Point>` they'd need to convert it into an array (true) but IMO that's not a big problem because they can easily do that using `toArray()`. That being said, we can always add an overloaded method that accepts `List<Point>` if needed - I was trying to avoid "confusion" exposing multiple APIs doing the "same" 👀 https://github.com/mapbox/mapbox-navigation-android/pull/1230#issuecomment-416186949

BTW, I wouldn't worry too much (for now) because I believe the downstream APIs (`Navigator`) are going to change soon and we'll need to revisit anyways. cc @kevinkreiser